test/python/test_kinetics.py

```diff
@@ -1166,64 +1159,74 @@ def test_opposite_direction3(self):
 
     def test_opposite_direction4(self):
         gas = ct.Solution(self.infile, 'E')
-        self.assertEqual(gas.n_reactions, 2)
+        assert gas.n_reactions == 2
 
     def test_common_efficiencies(self):
         self.check('F')
 
     def test_disjoint_efficiencies(self):
         gas = ct.Solution(self.infile, 'G')
-        self.assertEqual(gas.n_reactions, 2)
+        assert gas.n_reactions == 2
 
     def test_different_type(self):
         gas = ct.Solution(self.infile, 'H')
-        self.assertEqual(gas.n_reactions, 2)
+        assert gas.n_reactions == 2
 
     def test_declared_duplicate(self):
         gas = ct.Solution(self.infile, 'I')
-        self.assertEqual(gas.n_reactions, 2)
+        assert gas.n_reactions == 2
 
     def test_unmatched_duplicate(self):
         self.check('J')
 
     def test_nonreacting_species(self):
         gas = ct.Solution(self.infile, 'K')
-        self.assertEqual(gas.n_reactions, 3)
+        assert gas.n_reactions == 3
+
 
+# Fixture for the following test classes
+@pytest.fixture(scope='class')
+def initial_conditions():
+    return {'T': 900, 'P': 2*ct.one_atm,
+            'X': 'H2:0.1, H2O:0.2, O2:0.7, O:1e-4, OH:1e-5, H:2e-5'}
 
-class TestReaction(utilities.CanteraTest):
-    @classmethod
-    def setUpClass(cls):
-        utilities.CanteraTest.setUpClass()
-        cls.gas = ct.Solution('h2o2.yaml', transport_model=None)
-        cls.gas.X = 'H2:0.1, H2O:0.2, O2:0.7, O:1e-4, OH:1e-5, H:2e-5'
-        cls.gas.TP = 900, 2*ct.one_atm
-        cls.species = ct.Species.list_from_file("h2o2.yaml")
+@pytest.fixture(scope='class')
+def gas(initial_conditions):
+    gas = ct.Solution('h2o2.yaml', transport_model=None)
+    gas.X = initial_conditions['X']
```

Comment:
I was originally separating them so that different conditions could in theory be parameterized and used, but I think that in most of the cases, the tests are very specific to those exact conditions.